CFC_WebApp/main/mode/cluster_pipeline.py

```diff
@@ -0,0 +1,327 @@
+import os, sys
+sys.path.append("%s/../" % os.getcwd())
+
+from route_matching import update_user_routeDistanceMatrix, update_user_routeClusters
+from K_medoid_2 import kmedoids, user_route_data
+import math
+from get_database import get_section_db,get_profile_db, get_groundClusters_db, get_routeCluster_db
+import numpy as np
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+import pygmaps 
+from sklearn.cluster import KMeans
+from sklearn import manifold
+from route_matching import getRoute
+from util import read_uuids
+
+
+"""
+Notes
+
+Usage: python cluster_pipeline.py <username>
+
+Username must be associated with UUID in user_uuid.secret 
+
+High level overview:
+-This script provides a series of tools to help you evaluate your clustering algorithm across different methods of calculating distance.
+
+-For a particular user that you pass in to this script, we will generate and plot clusters on a 2d-plane using MDS. colors correspond
+to a kmedoid generated clusters
+
+-we also compare kmedoid generated clusters to ground truth clusters and returns accuracy score
+
+"""
+
+
+
+if not os.path.exists('mds_plots'):
+    os.makedirs('mds_plots')
+
+
+
+def extract_features(model_name, user_id, method=None, is_ground_truth=False):
+    data = None
+    if model_name == 'kmeans':
+        data = generate_section_matrix(user_id)
+        return data
+    elif model_name == 'kmedoid':
+        data = get_user_disMat(user_id, method=method, is_ground_truth=is_ground_truth)
+    return data
+
+def generate_clusters(model_name, data, user_id, method=None, is_ground_truth=False):
+    clusters = None
+    if model_name == 'kmeans':
+        clusters = kmeans(data)
+    elif model_name == 'kmedoid':
+        clusters = get_user_clusters(user_id, method=method, nClusters=-1, is_ground_truth=is_ground_truth)
+    return clusters
+
+def evaluate_clusters():
+    pass
+
+
+#########################################################################################################
+                                    # LOW LEVEL ABSTRACTION #
+#########################################################################################################
+
+
+def get_user_sections(user_id):
+    sections = list(get_section_db().find({'$and':[{'user_id': user_id},{'type': 'move'}]}))
+    return sections
+
+def get_user_disMat(user, method, is_ground_truth=False):
+    ## update route clusters:
+    print "Generating route clusters for %s" % user
+    if is_ground_truth:
```

Comment:
why does the distance matrix generated depend on whether it is ground truth or not? The distances are the same, right, only the colors of the labels change with ground truth or other clustering techniques.